docs: Document DragonflyDB support

SadikSunbul opened the pull request:
- This pull request introduces Dragonfly, a Redis-like storage solution, as an additional storage option for the Fiber framework.
- Dragonfly is a powerful alternative to Redis that offers similar functionality with potential improvements in scalability and performance.
- Dragonfly has been integrated as a storage option for Fiber, providing users with the flexibility to choose between Redis and Dragonfly for their caching and storage needs. 
- The integration is designed to be seamless and easy to implement, following the same interface used for Redis.

<img width="891" alt="Ekran Resmi 2025-01-30 18 09 38" src="https://github.com/user-attachments/assets/b6622735-574d-49b5-a596-3087397c4183" />


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Documentation**
	- Enhanced the documentation with guidance on using DragonflyDB as a seamless, fully compatible Redis alternative.
	- Added clear instructions in the installation section, enabling a plug-and-play experience without requiring code changes.
	- Updated comments for clarity in the Redis storage creation function.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

gaby commented:
@SadikSunbul I dont think we need this driver. You can talk to DragonflyDB using our existing Redis and Rueidis drivers

SadikSunbul commented:
> @SadikSunbul I dont think we need this driver. You can talk to DragonflyDB using our existing Redis and Rueidis drivers

You're right that Dragonfly can be used with existing Redis and Rueidis drivers. Instead of adding a separate driver, I can update the documentation to explicitly mention that Fiber’s Redis storage is compatible with Dragonfly. This would help users understand that they can use Dragonfly without any additional integration. Let me know if that sounds good to you!

SadikSunbul commented:
- Added DragonflyDB usage details to Redis README
- Removed the DragonflyDB file
- No extra tests were added for DragonflyDB; existing Redis tests work for it

gaby commented on redis/README.md:
```diff
@@ -33,6 +33,12 @@ func (s *Storage) Keys() ([][]byte, error)
 ### Installation
 Redis is tested on the 2 last [Go versions](https://golang.org/dl/) with support for modules. So make sure to initialize one first if you didn't do that yet:
+
+> **Note:** You can also use [DragonflyDB](https://dragonflydb.io/) as a Redis replacement.  
```
Remove trailing spaces.

gaby commented on redis/redis.go:
```diff
@@ -13,7 +13,11 @@ type Storage struct {
+//
+// DragonflyDB is fully compatible with the Redis API, meaning it can be used as a drop-in replacement for Redis.
+//
+// Because of this compatibility, DragonflyDB can also be initialized through this function.
```
I dont think we need this new comments here. The ones in the README are enough.

gaby commented on redis/config.go:
```diff
@@ -6,6 +6,8 @@ import (
 
 // Config defines the config for storage.
+//
+// Dragonfly and Redis share the same config
```
I dont think we need this new comments here. The ones in the README are enough.

gaby reviewed (approved):
LGTM
